tests/e2e/utils/pageObjects/wp/wpInitiativeEvaluation.page.ts

```diff
@@ -0,0 +1,50 @@
+import { expect, Locator, Page } from "@playwright/test";
+import BasePage from "../base.page";
+
+export class WPEvaluationPlanPage extends BasePage {
+  public path = "/wp/state-or-territory-specific-initiatives/initiatives";
+
+  readonly page: Page;
+  readonly title: Locator;
+  readonly backButton: Locator;
+  readonly addObjectiveButton: Locator;
+  readonly modal: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+    this.title = page.getByRole("heading", {
+      name: "State or Territory-Specific Initiatives: I. Define initiative",
+    });
+    this.backButton = page
+      .getByRole("button", {
+        name: "Return to dashboard for this initiative",
+      })
+      .first();
+    this.addObjectiveButton = page
+      .getByRole("button", {
+        name: "Add objective",
+      })
+      .first();
+    this.modal = page.getByRole("dialog");
+  }
+
+  public async fillFields() {
+    const modal = this.page.getByRole("dialog");
+    const textAreas = await modal.getByRole("textbox").all();
```

Comment:
Using `.all()` opens us up to possibly flaky tests (see [note in documentation](https://playwright.dev/docs/api/class-locator#locator-all)), because it immediately returns the locators without necessarily waiting for the page to load. Paradoxically, it appears to be the only way to [iterate over locators](https://playwright.dev/docs/locators#do-something-with-each-element-in-the-list), so you will see that I've added some `expects` in these files to ensure that we're triggering certain actions before moving on, in an effort to tell the test runner to slow its roll.

Reply:
Interesting! Since this is iterating through fields in a modal, after the modal has loaded, I think it's reasonable for us to expect all those fields to have loaded.

I don't know if that's necessarily guaranteed by React's update loop or anything, but we're very much _not_ doing extra fetches to asynchronously add inputs after the modal is open. So it seems low-risk to me 👍

Reply:
I suppose, if we ever do run into flakiness here, the answer would be to explicitly list all the textboxes' expected labels? And then iterate over those, waiting to find (& fill) an input with each label. I've forgotten exactly what this modal looks like, so I can't remember how difficult that would be - if we'd be able to hardcode some labels, or if we'd need to auto-generate them based on the report period or what have you.

In either case, I'm not too worried about this.